protocols/kad/src/kbucket.rs

```diff
@@ -30,65 +30,53 @@
 use arrayvec::ArrayVec;
 use bigint::U512;
 use multihash::Multihash;
-use parking_lot::{Mutex, MutexGuard};
 use std::mem;
-use std::slice::Iter as SliceIter;
+use std::slice::IterMut as SliceIterMut;
 use std::time::{Duration, Instant};
 use std::vec::IntoIter as VecIntoIter;
 
 /// Maximum number of nodes in a bucket.
 pub const MAX_NODES_PER_BUCKET: usize = 20;
 
-/// Table of k-buckets with interior mutability.
-#[derive(Debug)]
+/// Table of k-buckets.
+#[derive(Debug, Clone)]
 pub struct KBucketsTable<Id, Val> {
+    /// Peer ID of the local node.
     my_id: Id,
-    tables: Vec<Mutex<KBucket<Id, Val>>>,
+    /// The actual tables.
+    tables: Vec<KBucket<Id, Val>>,
     // The timeout when pinging the first node after which we consider that it no longer responds.
     ping_timeout: Duration,
 }
 
-impl<Id, Val> Clone for KBucketsTable<Id, Val>
-where
-    Id: Clone,
-    Val: Clone,
-{
-    #[inline]
-    fn clone(&self) -> Self {
-        KBucketsTable {
-            my_id: self.my_id.clone(),
-            tables: self
-                .tables
-                .iter()
-                .map(|t| t.lock().clone())
-                .map(Mutex::new)
-                .collect(),
-            ping_timeout: self.ping_timeout.clone(),
-        }
-    }
-}
-
+/// An individual table that stores peers or values.
 #[derive(Debug, Clone)]
 struct KBucket<Id, Val> {
-    // Nodes are always ordered from oldest to newest.
-    // Note that we will very often move elements to the end of this. No benchmarking has been
-    // performed, but it is very likely that a `ArrayVec` is the most performant data structure.
+    /// Nodes are always ordered from oldest to newest.
+    /// Note that we will very often move elements to the end of this. No benchmarking has been
+    /// performed, but it is very likely that a `ArrayVec` is the most performant data structure.
     nodes: ArrayVec<[Node<Id, Val>; MAX_NODES_PER_BUCKET]>,
 
-    // Node received when the bucket was full. Will be added to the list if the first node doesn't
-    // respond in time to our ping. The second element is the time when the pending node was added.
-    // If it is too much in the past, then we drop the first node and add the pending node to the
-    // end of the list.
+    /// Node received when the bucket was full. Will be added to the list if the first node doesn't
```

Comment:
```suggestion
     /// Nodes received when the bucket was full will be added to the list if the first node doesn't
```

Reply:
It's only a single node. What this comment says is that the content of `pending_node` can be added to `nodes` at a later point in time.